multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/clients/CustomServiceKeysClient.java

```diff
@@ -95,58 +48,37 @@ public List<DeployedMtaServiceKey> getServiceKeysByMetadataAndGuids(String space
 
     private List<DeployedMtaServiceKey> getServiceKeysByMetadataInternal(String labelSelector, List<DeployedMtaService> services) {
         String uriSuffix = includeServiceInstanceResourcesParam;
-        List<DeployedMtaService> detectedServices = getOnlyPopulatedServices(services);
-        if (detectedServices != null) {
-            uriSuffix = detectedServices.stream()
-                                .map(service -> service.getGuid()
-                                                       .toString())
-                                .collect(Collectors.joining(","));
-            uriSuffix = "&service_instance_guids=" + uriSuffix;
+        List<DeployedMtaService> managedServices = getManagedServices(services);
+        if (managedServices != null) {
+            uriSuffix += "&service_instance_guids=" + managedServices.stream()
+                                                                     .map(service -> service.getGuid()
+                                                                                            .toString())
+                                                                     .collect(Collectors.joining(","));
         }
-
-        return getListOfResources(new ServiceKeysResponseMapper(detectedServices), serviceKeysByMetadataSelectorUri + uriSuffix,
+        return getListOfResources(new ServiceKeysResponseMapper(managedServices), serviceKeysByMetadataSelectorUri + uriSuffix,
                                   labelSelector);
     }
 
-    private List<DeployedMtaService> getOnlyPopulatedServices(List<DeployedMtaService> services) {
-        if (services == null)
+    private List<DeployedMtaService> getManagedServices(List<DeployedMtaService> services) {
+        if (services == null) {
             return null;
-
-        List<DeployedMtaService> populatedServices = services.stream()
-                                                             .filter(this::serviceIsNotUserProvided)
-                                                             .collect(Collectors.toList());
-        
-        return populatedServices.isEmpty() ? null : populatedServices;
-    }
-
-    private boolean serviceIsNotUserProvided(DeployedMtaService service) {
```

Comment:
The original code was redundant. It was checking if service objects had GUIDs, which they will always have since we are getting them from the cloud controller. It is more appropriate to check the service type, since service keys can only be created for managed service instances.